docs: add guide for managing short URLs on AsyncAPI website (#4491)
asyncapi/website: README.md

A case study becomes publicly available right after merging and rebuilding the website.

> # 🌐 Adding Short URLs to the AsyncAPI Website

Reviewer: This advice is worth looking into @vishvamsinh28

> # 🌐 Adding Short URLs to the AsyncAPI Website

Reviewer 2: Never use emojis in headings, they break anchor urls.

Please [check the reply in another PR with the same issue](https://github.com/asyncapi/website/pull/4457#discussion_r2445687267)

Short URLs are useful for:
- Community resources (Slack, GitHub)
- Frequently accessed documentation
- Marketing campaigns and events

---

## ⚙️ How It Works

All redirects are defined in the `public/_redirects` file. This file follows **Netlify’s redirect syntax** and is automatically processed during deployment.

### Existing Examples

| Short URL | Redirects To | Type |
|------------|---------------|------|
| `/slack-invite` | Slack invitation link | Temporary (302) |
| `/modelina` | `/tools/modelina` | Permanent (301) |
| `/generator` | `/tools/generator` | Permanent (301) |
| `/cheatsheet` | GitHub cheatsheet | Permanent (301) |

---

## ✅ When to Create a Short URL

- **Community resources:** Slack, GitHub repositories  
- **Frequently accessed tools:** AsyncAPI Studio, Generator, Modelina  
- **Docs shortcuts:** Specification, Getting Started, Tutorials   
- **Events & campaigns:** Conferences

### Avoid Short URLs For

- Frequently changing internal pages  
- Temporary content without clear end dates  
- URLs that conflict with existing site routes  
- Personal or private organization links  

> - Already short, self-explanatory URLs  

Reviewer 2: I would move these sections above instead of lines 226-229, as they repeat the same information

> ---

Reviewer 2: ```suggestion
```

## 🪜 How to Add a New Short URL

### Step 1: Plan

1. **Identify the target URL** you want to redirect to.  

> 2. **Choose a descriptive short path** (see naming conventions below).  

Reviewer 2: > see naming conventions below

Where? 🤔

Author: it was above not below 😅

> 3. **Select the correct redirect type** — permanent (301) or temporary (302).  

Reviewer 2: ```suggestion
3. **Select the appropriate redirect type**, for example, permanent (301) or temporary (302).  
```

4. **Check for conflicts** with existing routes or redirects.

> ### Step 2: Edit the Redirects File

Reviewer 2: Some examples would be helpful here

1. Navigate to `public/_redirects`.  
2. Find or create an appropriate section (e.g., “Community Resources”).  
3. Add your redirect entry using the correct syntax.  
4. Include a descriptive comment explaining the redirect’s purpose.

### Step 3: Submit and Test

1. **Create a Pull Request (PR)** with your changes.  

> 2. Add **context in the PR description** — why this short URL is needed.  

Reviewer 2: ```suggestion
2. Add **context in the PR description** describing why this short URL is needed.  
```

> 3. **Test the redirect** after deployment to staging/production.  

Reviewer 2: Do we have staging env?

> 4. **Monitor for issues** and update if necessary.

Reviewer 2: Does it mean that the author of a new short url will forever be its maintainer?

---

## 🧩 Redirect Syntax

> `/short-path destination-url status-code!`

Reviewer 2: This section should be a part of the step 2 of the guide above
